### ISSUE TITLE
[Tech Debt] Refactor the anonymous function inside secretstore-setup for init Vault

### ISSUE THREAD

**Issue by jim-wang-intel** [edgexfoundry/edgex-go#3182]
# 🚀 Feature Request

### Relevant Package

 
This feature request is for  `secretstore-setup`. 

### Description
 
Currently, there is big anonymous function in the `secretstore-setup` to initialize and unseal Vault.  That part does n't have any tests to verify the functionality.  As being very critical part of that service, we should add unit tests for that function.

### Describe the solution you'd like
 
Refactor that portion of anonymous function out into a separate package and add unit tests to it using Vault's mocking client from go-mod-secret module.  
Hook it up inside the secretstore-setup as anonymous func using this refactored new functions in new package.

### Describe alternatives you've considered
 Have you considered any alternative solutions or workarounds?

**Comment by bnevis-i** [comment]
Non-breaking change, internal design.  Leaving in inbox. Last reviewed 5/26/2021.
